dd-sdk-android/src/main/kotlin/com/datadog/android/tracing/internal/domain/SpanSerializer.kt

```diff
@@ -145,6 +146,11 @@ internal class SpanSerializer(
         if (!userInfo.email.isNullOrEmpty()) {
             jsonLog.addProperty(LogAttributes.USR_EMAIL, userInfo.email)
         }
+        // add extra attributes
+        userInfo.extraInfo.forEach {
+            val key = "${LogAttributes.USR_ATTRIBUTES_GROUP}.${it.key}"
+            jsonLog.add(key, it.value.toJsonElement())
```

Comment:
Here we encode info under `span.meta.*` key. The APM Intake expects all values encoded under this object to be `String`. That's why in `addLogNetworkInfo()` we call `.toString()` for each encoded value. I think you need to also call it there, otherwise all `Spans` with custom user info value other than `String` will be rejected, e.g. setting this:
```kotlin
Datadog.setUserInfo("123", "foo", "foo@bar.com", ["is-registered": true])
```
will probably lead to all `Spans` being rejected.

cc @xgouchet @buranmert

Reply:
hmm...good catch, I am going to test it and see what happens if we send a non string value there. In case is being rejected I am going to cast it to a String.